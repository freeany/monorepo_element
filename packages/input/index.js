import RrInput from './src/input.vue'

RrInput.install = Vue => {
  Vue.component(RrInput.name, RrInput)
}

export default RrInput
