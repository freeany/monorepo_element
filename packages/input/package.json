{
  "name": "rr-input",
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "gitHead": "de5f4f94ab442c88e6da53702b59c99392e0957d"
}
