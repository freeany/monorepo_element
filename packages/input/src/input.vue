<template>
  <div>
    <input :type="type" :value="value" v-bind="$attrs" @input="inputHandle" />
  </div>
</template>

<script>
export default {
  name: 'RrInput',
  inheritAttrs: false,
  props: {
    type: {
      type: String,
      default: 'text'
    },
    value: {
      type: String,
      default: ''
    }
  },
  methods: {
    inputHandle(e) {
      this.$emit('input', e.target.value)
      // let hasFormItem = ''
      const searchFormItem = parent => {
        if (!parent) return
        if (parent.$options.name !== 'LgFormItem') {
          parent = parent.$parent
          return searchFormItem(parent)
        } else {
          // hasFormItem = parent
          return parent
        }
      }
      const formItem = searchFormItem(this.$parent)
      formItem && formItem.validate() // 也可以使用发布订阅模式， 在此发布，在formItem中订阅，并调用formItem的validate方法，区别是啥？
    }
  }
}
</script>

<style></style>
