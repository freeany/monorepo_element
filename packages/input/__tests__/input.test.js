import input from '../src/input.vue'
import { mount } from '@vue/test-utils'

// rr-input是名字
describe('rr-input', () => {
  // 测试函数
  test('input-text', () => {
    // 现在内存中进行挂载input组件
    const wrapper = mount(input)
    // 判断组件生成的html是否包含 <input type="text" />
    // wrapper.html() 得到的是<div><input type=\"text\"></div>
    expect(wrapper.html()).toContain('input type="text"')
  })

  // 测试密码框
  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.html()).toContain('input type="password')
    // 测试传递的值是否是admin
    expect(wrapper.props('value')).toBe('admin')
  })

  // 测试快照, 存储快照...测试的快照会存放到当前文件夹下的__snapshots__/input.test.js.snap
  // yarn test -u 重新生成快照
  test('input-snapshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
