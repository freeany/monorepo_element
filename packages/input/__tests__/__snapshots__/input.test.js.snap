// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rr-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
